### PR TITLE
Show

### PR DESCRIPTION
**Pull request by kalmarek**
+ some predicates

### CODE REVIEW

File: src/predicates.jl [hunk 1]
```diff
@@ -46,3 +46,66 @@ for (T, funcpairs) in (
...
+            res = ccall(@libarb($arbf), Cint, (Ref{Arf}, $ctype), x, y)
+            return res < 0
+        end
+        Base.:(<)(x::Arf, y::$jltype) = isless(x, y)
+        Base.:(<=)(x::Arf, y::$jltype) = isequal(x,y) | isless(x,y)
+    end
+end
+
+for (ArbT, args) in (
+    (Arb, (
```
Comment (saschatimme) [inline src/predicates.jl]: Why not also for Arf?

File: src/constructors.jl [hunk 2]
```diff
@@ -75,26 +81,32 @@ for ArbT in (:Arf, :Arb, :Acb, :Mag)
...
 
-Arf(x::BigFloat, prec::Integer) = set!(Arf(prec), x)
-Arb(x::BigFloat, prec::Integer) = set!(Arb(prec), Arf(x, prec))
-Acb(x::BigFloat, prec::Integer) = set!(Acb(prec), Arb(x, prec))
+Arf(x::BigFloat, prec::Integer=precision(x)) = set!(Arf(prec), x)
+Arb(x::BigFloat, prec::Integer=precision(x)) = set!(Arb(prec), Arf(x, prec))
+Acb(x::BigFloat, prec::Integer=precision(x)) = set!(Acb(prec), Arb(x, prec))
 
 # fallbacks:
 Arf(x::Real, prec::Integer) = set!(Arf(prec), BigFloat(x))
```
Comment (saschatimme) [inline src/constructors.jl]: I think we should still have a global `precision` variable as a fallback. Just if you want to be lazy and just quickly try something.

File: src/predicates.jl (hunk 1 above)
Comment (kalmarek) [reply src/predicates.jl]: there are no `arf_eq`, etc. `libarb` implements only `arf_equal` and `arf_cmp*`

File: src/constructors.jl (hunk 2 above)
Comment (kalmarek) [reply src/constructors.jl]: ok, but what about thread safeness?

Comment (saschatimme) [reply src/constructors.jl]: Sure, this is not sensible for thread safe code, but you also do not write hastily thread safe code 

I am just thinking of defining
```
const DEFAULT_PREC = Ref(256)
precision(x::Number) = DEFAULT_PREC[]
```

In library code, one definitely should always carry around the precision

**Pull request merged**